pkg/services/secrets/kvstore/remote_plugin.go

```diff
@@ -51,6 +63,8 @@ func (kv *secretsKVStorePlugin) Set(ctx context.Context, orgId int64, namespace
 		err = wrapUserFriendlySecretError(res.UserFriendlyError)
 	}
 
+	updateFatalFlag(ctx, *kv)
```

Comment:
Couldn't migration handle this flag entirely? Plugin migration would run even if there's nothing to migrate, the flag could be set then based on the compatibility flag before iterating over secrets, so it would be set even if there was nothing to migrate.

Isn't migration already handling this? Why do you need the `sync.Once` or the `backwardsCompatibilityDisabled` on the struct or the `updateFatalFlag` function, does this need to be set again after startup?